### ISSUE TITLE
loss does not decrease in Google Colab and local computer. (TF 1.13 vs  from TF 1.14 to TF 2.6)

### ISSUE THREAD

**Issue by HeewonChung92** [tensorflow/tensorflow#56080]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Performance

### Source

source

### Tensorflow Version

1.13 vs 2.6

### Custom Code

Yes

### OS Platform and Distribution

Windows 10

### Python version

3.7.4

### CUDA/cuDNN version

11.2.0 / 8.1.0

### Current Behaviour?

```shell
I studied with my data and tf-gpu 1.6 and keras in 2019. 
At this time, training was done normally.

Although it is the same code as in the past, 
training loss and validation loss are decreased 
when training with TF 1.13  (working TF 1.6 ~ TF 1.13).

However, from TF 1.14 to TF 2.6, 
both train loss and validation loss have increased.

When the model weight made with TF 1.6 was called from TF 2.6 (tried to TF 1.14, TF 1.15, TF 2.0 and TF 2.6), 
it was called without any problems and the results were the same as those from the past.

So I can't understand why that happen.
I wonder what parts of TF 1.13 and TF 1.14 and above are different and affect training!!!!!!

Also, I tried updating the source code through tf_upgrade_v2, 
but it was not solved at all.
https://www.tensorflow.org/guide/upgrade?hl=en
tf_upgrade_v2 --infile my_code.py --outfile my_code_upgraded.py
```

### Standalone code to reproduce the issue

```shell
def define_model(time_steps, input_rows, input_dim, input_ch, output_dim, dropRate, learning_rate):   
    input_data = tf.keras.layers.Input(shape=(time_steps, input_rows, input_dim, input_ch), name='my_data')

    x = tf.keras.layers.TimeDistributed(tf.keras.layers.Conv2D(filters=16, kernel_size=(2, 2), strides=4, padding='same'))(input_data )
    x = tf.keras.layers.LeakyReLU()(x)   # default alpha = 0.3
    x = tf.keras.layers.TimeDistributed(tf.keras.layers.MaxPooling2D((2, 2), strides=2))(x)
    x = tf.keras.layers.Dropout(rate=dropRate)(x)

    x = tf.keras.layers.TimeDistributed(tf.keras.layers.Flatten())(x)
    x = tf.keras.layers.TimeDistributed(tf.keras.layers.Dense(128))(x)
    x = tf.keras.layers.LeakyReLU()(x)   # default alpha = 0.3

    ###============================== Model 2 (LSTM)
    x = tf.keras.layers.LSTM(128, dropout=0.2, recurrent_dropout=0.2, return_sequences=True)(x)
    x = tf.keras.layers.LSTM(10, dropout=0.2, recurrent_dropout=0.2, return_sequences=False)(x)

    x_out = tf.keras.layers.Dense(10, 'softmax')(x)

    model = Model(inputs=input_data, outputs=x_out)
    model.compile(optimizer=optimizers.Adam(learning_rate=0.001), loss=losses.categorical_crossentropy)

    return model
```

### Relevant log output

**decreased the loss when using TF 1.13**
![image](https://user-images.githubusercontent.com/39148033/168065098-12939b28-dbe7-4cef-9cb3-ecb0ea18843c.png)

**increased the loss when using TF 1.14 and TF 2.6**
![image](https://user-images.githubusercontent.com/39148033/168066373-cb5bb8b4-1f18-41ba-a412-446c5d0dd195.png)

_No response_</details>

**Comment by sushreebarsa** [comment]
@HeewonChung92 
In order to expedite the trouble-shooting process, please provide a code snippet to reproduce the issue reported here. Thank you!

**Comment by HeewonChung92** [comment]
Hello, @sushreebarsa !!!
Thank you very much for your help.

This is my code!
https://colab.research.google.com/drive/1tSE7t3GPa8r4x7F4sPgj8CqiYqtOrej5?usp=sharing
https://drive.google.com/drive/folders/17Nnl5FcrUWnRgzz4DM_fNAI3KhwqcK3r?usp=sharing

**Comment by HeewonChung92** [comment]
@sushreebarsa 
Yes, that's same code on colab!
I tested to your code on colab and it makes the same issue.
 
As you installed it, I tested it by installing various versions in the same way. 
(!pip install tensorflow-gpu==1.13 or 1.14 or 2.6)
Depending on the version of tensorflow, it shows different learning rates when training the data (increased or decreased).

The important thing about my training is that the loss and error are lowered. (Accuracy can be ignored.)
